Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py

```diff
@@ -0,0 +1,406 @@
+
+IPAddressRegex = re.compile(
+    r"((^\s*((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4]\""
+    "[0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))"
+    "\s*$)|(^\s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f"
+    "{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25"
+    "[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:"
+    "[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5"
+    "|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:"
+    "[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]\d|"
+    "1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(("
+    "[0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4})"
+    "{0,2}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|"
+    "[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5}|"
+    "((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]"
+    "|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:"
+    "[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]\d|"
+    "1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(:(((:"
+    "[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]\d|"
+    "1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:)))(%.+)?"
+    "\s*$))\")"
+)
+
+
+def validateIP(ip):
+    if (re.match(IPAddressRegex, str(ip))):
+        return True
+    else:
+        demisto.debug("Invalid IP Address: " + str(ip))
+        return False
+
+
+urlregex = re.compile(
+    r'^https?://'  # http:// or https://
+    r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+[A-Z]{2,6}\.?|'  # domain...
+    r'localhost|'  # localhost...
+    r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'  # ...or ip
+    r'(?::\d+)?'  # optional port
+    r'(?:/?|[/?]\S+)$', re.IGNORECASE)
+
+
+def validateUrl(url):
+    if(re.match(urlregex, str(url))):
+        return True
+    else:
+        demisto.debug("Invalid Url: " + str(url))
+        return False
+
+
+emailregex = '^[a-zA-Z0-9._%+-]+@.*[.]\w{2,3}$'
+
+
+def validateEmail(email):
+    if(re.match(emailregex, str(email))):
+        return True
+    else:
+        demisto.debug("Invalid Email: " + str(email))
+        return False
+
+
+def cleanData(uncleanListOfIndicators):
+
+    cleanListOfIndicators = []
+
+    # Iterate through the unclean list of indicators
+    for i in uncleanListOfIndicators:
+        indicator = i  # .to_dict(True)
+        itype = indicator['indicatorType'].lower()
+        demisto.info("Indicator: " + str(indicator) + " - indicator_type (lowercase): " + itype)
+
+        # Chcks to see if indicator already exists
+        if i not in cleanListOfIndicators:
+
+            # Make calls to the check functions for URLs, emails, IPv4 addresses, IPv6 addresses
+            if itype == 'ipv4' or itype == 'ipv6' or itype == 'ip':
+                if validateIP(indicator['value']) is True:
```

Comment:
What will happen here (as well as in other places in this function) if  `value` does not exist in `indicator`?
My suggestion is to use `get` (for avoiding an unexpected exception):
```suggestion
                if validateIP(indicator.get('value')) is True:
```